x-pack/plugins/ml/public/register_helper/register_search_links/search_deep_links.ts

```diff
@@ -8,203 +8,286 @@
+  return {
+    id: 'dataVisualizer',
+    title: i18n.translate('xpack.ml.deepLink.dataVisualizer', {
+      defaultMessage: 'Data Visualizer',
+    }),
+    path: `/${ML_PAGES.DATA_VISUALIZER}`,
+    navLinkStatus:
+      mlCapabilities.isADEnabled && mlCapabilities.isDFAEnabled && mlCapabilities.isNLPEnabled
+        ? AppNavLinkStatus.visible
+        : AppNavLinkStatus.hidden,
+  };
+}
+
+function getFileUploadDeepLink(mlCapabilities: MlCapabilities): AppDeepLink<LinkId> {
+  return {
+    id: 'fileUpload',
+    title: i18n.translate('xpack.ml.deepLink.fileUpload', {
+      defaultMessage: 'File Upload',
+    }),
+    keywords: ['CSV', 'JSON'],
+    path: `/${ML_PAGES.DATA_VISUALIZER_FILE}`,
+    navLinkStatus:
+      mlCapabilities.isADEnabled && mlCapabilities.isDFAEnabled && mlCapabilities.isNLPEnabled
+        ? AppNavLinkStatus.visible
+        : AppNavLinkStatus.hidden,
+  };
+}
+
+function getIndexDataVisualizerDeepLink(mlCapabilities: MlCapabilities): AppDeepLink<LinkId> {
+  return {
+    id: 'indexDataVisualizer',
+    title: i18n.translate('xpack.ml.deepLink.indexDataVisualizer', {
+      defaultMessage: 'Index Data Visualizer',
+    }),
+    path: `/${ML_PAGES.DATA_VISUALIZER_INDEX_SELECT}`,
+    navLinkStatus:
+      mlCapabilities.isADEnabled && mlCapabilities.isDFAEnabled && mlCapabilities.isNLPEnabled
+        ? AppNavLinkStatus.visible
+        : AppNavLinkStatus.hidden,
+  };
+}
+
+function getSettingsDeepLink(mlCapabilities: MlCapabilities): AppDeepLink<LinkId> {
+  return {
+    id: 'settings',
+    title: i18n.translate('xpack.ml.deepLink.settings', {
+      defaultMessage: 'Settings',
+    }),
+    path: `/${ML_PAGES.SETTINGS}`,
+    navLinkStatus: mlCapabilities.isADEnabled ? AppNavLinkStatus.visible : AppNavLinkStatus.hidden,
+    deepLinks: [
+      {
+        id: 'calendarSettings',
+        title: i18n.translate('xpack.ml.deepLink.calendarSettings', {
+          defaultMessage: 'Calendars',
+        }),
+        path: `/${ML_PAGES.CALENDARS_MANAGE}`,
+        navLinkStatus: mlCapabilities.isADEnabled
+          ? AppNavLinkStatus.visible
+          : AppNavLinkStatus.hidden,
+      },
+      {
+        id: 'filterListsSettings',
+        title: i18n.translate('xpack.ml.deepLink.filterListsSettings', {
+          defaultMessage: 'Filter Lists',
+        }),
+        path: `/${ML_PAGES.SETTINGS}`, // Link to settings page as read only users cannot view filter lists.
+        navLinkStatus: mlCapabilities.isADEnabled
+          ? AppNavLinkStatus.visible
+          : AppNavLinkStatus.hidden,
+      },
+    ],
+  };
+}
+
+function getNotificationsDeepLink(mlCapabilities: MlCapabilities): AppDeepLink<LinkId> {
+  return {
+    id: 'notifications',
+    title: i18n.translate('xpack.ml.deepLink.notifications', {
+      defaultMessage: 'Notifications',
+    }),
+    path: `/${ML_PAGES.NOTIFICATIONS}`,
+    navLinkStatus: AppNavLinkStatus.visible,
```

Comment:
shall we add an appropriate capabilities check for the notifications page?

Reply:
The notification page is always available in serverless and non-serverless modes.